src/token.rs

```diff
@@ -19,6 +19,7 @@ impl<C: TreeConfig> SyntaxToken<C> {
     #[inline(always)]
     pub(crate) unsafe fn new(idx: u32, tree_id: u32) -> Self {
         Self {
+            // I am wondering whether std::hint::unreachable_unchecked panics in debug?
```

Comment:
I just checked and nope, it doesn’t, so I guess it’s better to keep the check here in debug builds.